src/Compilers/Core/MSBuildTask/GenerateMSBuildAnalyzerConfig.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    public class GenerateMSBuildAnalyzerConfig : Task
+    {
+        [Output]
+        public string ConfigFileContents { get; set; }
+
+        [Required]
+        public ITaskItem[] MetadataItems { get; set; }
+
+        [Required]
+        public ITaskItem[] PropertyItems { get; set; }
+
+        public GenerateMSBuildAnalyzerConfig()
+        {
+            ConfigFileContents = string.Empty;
+            MetadataItems = new ITaskItem[0];
+            PropertyItems = new ITaskItem[0];
+        }
+
+        public override bool Execute()
+        {
+            StringBuilder builder = new StringBuilder();
+
+            // we always generate global configs
+            builder.AppendLine("is_global = true");
+
+            // collect the properties into a global section
+            foreach (var prop in PropertyItems)
+            {
+                builder.Append("msbuild_property.")
+                       .Append(prop.ItemSpec)
+                       .Append(" = ")
+                       .AppendLine(prop.GetMetadata("Value")); //TODO: do we need to encode/escape this?
```

Comment:
>TODO [](start = 65, length = 4)

I agree that there is an encoding/escaping question here. But we don't add TODO markers in the code. Some options: (1) we resolve this question, or (2) add a check (fail gracefully), or (3) create a tracking issue (link here).

Reply:
Good catch, I didn't notice that was still even in there. I'll write some tests to verify the behavior and either fix or open an issue.

---
In reply to: [419574615](https://github.com/dotnet/roslyn/pull/43617#discussion_r419574615) [](ancestors = 419574615)

Reply:
Filed [https://github.com/dotnet/roslyn/issues/43970](https://github.com/dotnet/roslyn/issues/43970) to track, and added a test that covers it.

---
In reply to: [419782382](https://github.com/dotnet/roslyn/pull/43617#discussion_r419782382) [](ancestors = 419782382,419574615)